### PR TITLE
WIP: Documenting email

### PR DESCRIPTION
**Pull request by kyzh**
This PR aim to provide a reference for people that set up a new instance of mastodon.
Email is not a big part of mastodon, but it can be quite difficult to know where to look and what to change.

This is a work in progress, I'd love to get feedback.

There is grammar and orthography issues as well as formatting issue.
My understanding of the technical details are likely not accurate too.
Feel free to point any of these issue as well as what I forgot to include here.
This is the documentation is was talking about in #66

### CODE REVIEW

File: email/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Email
+=====
+
+**Mastodon** uses email to get in touch with an account holder for different purposes.
+Mastodon uses outbound email only and do not process any inbound email.
+Instance administrator can also avoid using email altogether by confirming users themselves.
+This document highlight what emails are used for and where to change email configuration.  
+To go further,this document describe the default component involved from mastodon, rails and the OS and how they interact together.
+ 
+### What are email used for in Mastodon?
```
Comment (bmann) [inline email/README.md]: Should be: What "is" email used for?

File: email/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+In the default case, email will be first sent to confirm your user.
+Users can also choose to receive notification for various types of activity.
+Finally password reset and password reset confiration is done via email.
```
Comment (bmann) [inline email/README.md]: "confirmation"

**Comment by haliphax** [comment]
As outlined in tootsuite/mastodon issue [#2670](https://github.com/tootsuite/mastodon/issues/2670), it would be helpful to take docker into account with this, as well. Since the docker-compose setup does not include any sort of mail server, the default of attempting to connect to localhost fails with both SMTP and sendmail. Additionally, if SMTP is pointed to the correct host (through the docker network's gateway IP or to the localhost's public IP), the MTA must be configured to allow relaying from the docker network's subnet.

**Comment by dariusk** [comment]
Thank you so much for this PR but unfortunately it's a bit out of the scope of the Mastodon documentation. It's great to cover things like email architecture and go into details about SMTP, DKIM, etc, but that would probably be best for a separate blog post. We'll soon be adding a page with links to helpful external tutorials/resources and if you get it up as a blog post we'd love to link it from there.

**Comment by dariusk** [comment]
Also, I made some fixes to your grammar/spelling while reviewing this PR. Here's the commit as a PR to your email documentation branch, maybe you can find it helpful: https://github.com/kyzh/documentation/pull/1/commits/90642dfa4d8dd7d81dcf9734edb79a437ffd4607

**Pull request closed without merging**